### PR TITLE
Reset/Forgot  passsword for okta account

### PR DESCRIPTION
**Pull request by vurankar**
Add switch to reset forgot password for local and Okta account. 
Send forgot password email. 
Remove check for Accounts.LoginCancelledError.numericError login failure.

### CODE REVIEW

File: server/entry.coffee [hunk 1]
```diff
@@ -72,10 +72,36 @@ Meteor.startup ->
 
       #Send local user activation email.
       #specially useful when there is no internet connection
-      if process.env.USE_IDP == "local"
+      if Meteor.settings.public.useIDP == "local"
         Accounts.sendEnrollmentEmail(userId, user.email)
       # if (user.email && Accounts._options.sendVerificationEmail)
```
Comment (jonarnaldo) [inline server/entry.coffee]: get rid of this if not using

File: server/entry.coffee [hunk 2]
```diff
@@ -72,10 +72,36 @@ Meteor.startup ->
...
       #   Meteor.defer ->
       #     console.log("Send Verification Email")
       #     Accounts.sendVerificationEmail(userId, user.email)
       #     console.log("Verification Email Sent")
+
+    ###
+    TODO: move this method to the new package along with accounts-entry migration
```
Comment (jonarnaldo) [inline server/entry.coffee]: :+1:

File: server/entry.coffee [hunk 3]
```diff
@@ -72,10 +72,36 @@ Meteor.startup ->
...
+    ###
+    entryForgotPassword: (email) ->
+      if not email
+        throw new Meteor.Error(500, "Email cannot be blank")
+
+      syncForgotPassword = Meteor.wrapAsync(OktaClient.forgotPassword)
+      resp = syncForgotPassword email
+
+      user = Meteor.users.findOne({"emails.address":email});
```
Comment (jonarnaldo) [inline server/entry.coffee]: add space after ':' like so:
```'emails.address': email```

File: server/entry.coffee [hunk 4]
```diff
@@ -72,10 +72,36 @@ Meteor.startup ->
...
+
+      if !resp or !user
+        throw new Meteor.Error(500, "User not found")
+
+      link = resp.resetPasswordUrl + "?fromURI=" + Meteor.absoluteUrl()
```
Comment (jonarnaldo) [inline server/entry.coffee]: coffeescript has string interpolation:
```
 "#{resp.resetPasswordUrl}?fromURI=#{Meteor.absoluteUrl()}"
```

File: server/entry.coffee (hunk 1 above)
Comment (vurankar) [reply server/entry.coffee]: Done

File: server/entry.coffee (hunk 3 above)
Comment (vurankar) [reply server/entry.coffee]: done

**Pull request merged**